Rover/sailboat.cpp

```diff
@@ -239,18 +246,36 @@ void Sailboat::get_throttle_and_mainsail_out(float desired_speed, float &throttl
     if (!is_positive(desired_speed)) {
         mainsail_out = 100.0f;
     } else {
-        // Sails are sheeted the same on each side use abs wind direction
+        switch (sail_control_type) {
+
+            // standard linear interpolation control
+            case (SailControlType::LINEAR): {
+
+                // Sails are sheeted the same on each side use abs wind direction
+
+                // set the main sail to the ideal angle to the wind
+                float mainsail_angle = wind_dir_apparent_abs - sail_angle_ideal;
 
-        // set the main sail to the ideal angle to the wind
-        float mainsail_angle = wind_dir_apparent_abs - sail_angle_ideal;
+                // make sure between allowable range
+                mainsail_angle = constrain_float(mainsail_angle, sail_angle_min, sail_angle_max);
 
-        // make sure between allowable range
-        mainsail_angle = constrain_float(mainsail_angle,sail_angle_min, sail_angle_max);
+                // linear interpolate mainsail value (0 to 100) from wind angle mainsail_angle
+                float mainsail_base = linear_interpolate(0.0f, 100.0f, mainsail_angle, sail_angle_min, sail_angle_max);
 
-        // linear interpolate mainsail value (0 to 100) from wind angle mainsail_angle
-        float mainsail_base = linear_interpolate(0.0f, 100.0f, mainsail_angle,sail_angle_min,sail_angle_max);
+                mainsail_out = constrain_float((mainsail_base + pid_offset), 0.0f ,100.0f);
+                break;
+            }
+
+            // fixed angle strategy
+    		case (SailControlType::FIXED): {
 
-        mainsail_out = constrain_float((mainsail_base + pid_offset), 0.0f ,100.0f);
+                // linear interpolate mainsail value (0 to 100) from sail_angle_min
+    			float mainsail_base = linear_interpolate(0.0f, 100.0f, sail_angle_min, 0.0f, 90.0f);
+
+                mainsail_out = constrain_float((mainsail_base + pid_offset), 0.0f ,100.0f);
```

Comment:
This line can be outside the switch, it the same in both cases. Have to declare the mainsail_base before the switch.